src/emitter/diff.rs

```diff
@@ -25,11 +29,96 @@ impl Emitter for DiffEmitter {
         const CONTEXT_SIZE: usize = 3;
         let mismatch = make_diff(&original_text, formatted_text, CONTEXT_SIZE);
         let has_diff = !mismatch.is_empty();
-        print_diff(
-            mismatch,
-            |line_num| format!("Diff in {} at line {}:", filename, line_num),
-            &self.config,
-        );
+
+        if has_diff {
+            if self.config.print_misformatted_file_names() {
+                if self.num_reformatted_files < 1 {
+                    writeln!(output, "The below file(s) would be reformatted:")?;
```

Comment:
I prefer not to emit this header line, as it makes it harder to use the output of `cargo fmt -- --check -l` in pipelines or other scenarios.

Reply:
That's actually a relief, and I'm happy to remove it. Trying to come up with a header line that would work for both single/multiple files was more difficult than the actual work 😄

Reply:
Done!